src/android/java/io/jxcore/node/JXcoreExtension.java

```diff
@@ -512,7 +512,11 @@ public static void notifyPeerAvailabilityChanged(PeerProperties peerProperties,
 
         try {
             jsonObject.put(EVENT_VALUE_PEER_ID, peerProperties.getId());
-            jsonObject.put(EVENT_VALUE_PEER_GENERATION, peerProperties.getExtraInformation());
+            Integer gen = peerProperties.getExtraInformation();
+            if (!isAvailable) {
+                gen = (peerProperties.getExtraInformation() == 0) ? null : peerProperties.getExtraInformation();
```

Comment:
is it valid calling one method twice ? what about just cache value and check cached value?